app/src/main/java/net/gsantner/markor/format/zimwiki/ZimWikiLinkResolver.java

```diff
@@ -0,0 +1,115 @@
+package net.gsantner.markor.format.zimwiki;
+
+import net.gsantner.opoc.util.FileUtils;
+
+import org.apache.commons.io.FilenameUtils;
+
+import java.io.File;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Resolves zim links and converts them to file paths if necessary.
+ * The logic follows the specification of zim links as stated here: https://zim-wiki.org/manual/Help/Links.html
+ * If the link contains an additional description, it is parsed as well.
+ */
+public class ZimWikiLinkResolver {
+    private File _notebookRootDir;
+    private File _currentPage;
+
+    private String zimPath;
+    private String resolvedLink;
+    private String linkDescription;
+
+    public enum Patterns {
+        LINK(Pattern.compile("\\[\\[(?!\\[)((.+?)(\\|(.+?))?\\]*)]\\]")),
+
+        SUBPAGE_PATH(Pattern.compile("\\+(.*)")),
+        TOPLEVEL_PATH(Pattern.compile(":(.*)")),
+        RELATIVE_PATH(Pattern.compile("[^/]+")); // do not match weblinks
```

Comment:
I suggest to have all of these in the  ZimWikiHighlighter patterns - then they are all at one place.